overlord/state/fsbackend.go

```diff
@@ -0,0 +1,36 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package state
+
+import (
+	"github.com/ubuntu-core/snappy/osutil"
+)
+
+type StateFsBackend struct {
+	backFn string
+}
+
+func NewStateFsBackend(path string) *StateFsBackend {
+	return &StateFsBackend{backFn: path}
```

Comment:
Some suggestions:

* `StateFsBackend` => `fileBackend` (private)
* `NewStateFsBackend` => `NewFileBackend`, returning `StateBackend` (the interface)
* `backFn` => `path`